selfdrive/car/volkswagen/values.py

```diff
@@ -224,127 +231,219 @@ class CAR(Platforms):
+    chassis_codes={"5F"},
   )
-  SKODA_FABIA_MK4 = VolkswagenMQBPlatformConfig(  # Chassis PJ
+  SKODA_FABIA_MK4 = VolkswagenMQBPlatformConfig(
     [VWCarDocs("Škoda Fabia 2022-23", footnotes=[Footnote.VW_MQB_A0])],
     VolkswagenCarSpecs(mass=1266, wheelbase=2.56),
+    chassis_codes={"PJ"},
   )
-  SKODA_KAMIQ_MK1 = VolkswagenMQBPlatformConfig(  # Chassis NW
+  SKODA_KAMIQ_MK1 = VolkswagenMQBPlatformConfig(
     [
       VWCarDocs("Škoda Kamiq 2021-23", footnotes=[Footnote.VW_MQB_A0, Footnote.KAMIQ]),
       VWCarDocs("Škoda Scala 2020-23", footnotes=[Footnote.VW_MQB_A0]),
     ],
     VolkswagenCarSpecs(mass=1230, wheelbase=2.66),
+    chassis_codes={"NW"},
   )
-  SKODA_KAROQ_MK1 = VolkswagenMQBPlatformConfig(  # Chassis NU
+  SKODA_KAROQ_MK1 = VolkswagenMQBPlatformConfig(
     [VWCarDocs("Škoda Karoq 2019-23")],
     VolkswagenCarSpecs(mass=1278, wheelbase=2.66),
+    chassis_codes={"NU"},
   )
-  SKODA_KODIAQ_MK1 = VolkswagenMQBPlatformConfig(  # Chassis NS
+  SKODA_KODIAQ_MK1 = VolkswagenMQBPlatformConfig(
     [VWCarDocs("Škoda Kodiaq 2017-23")],
     VolkswagenCarSpecs(mass=1569, wheelbase=2.79),
+    chassis_codes={"NS"},
   )
-  SKODA_OCTAVIA_MK3 = VolkswagenMQBPlatformConfig(  # Chassis NE
+  SKODA_OCTAVIA_MK3 = VolkswagenMQBPlatformConfig(
     [
       VWCarDocs("Škoda Octavia 2015-19"),
       VWCarDocs("Škoda Octavia RS 2016"),
       VWCarDocs("Škoda Octavia Scout 2017-19"),
     ],
     VolkswagenCarSpecs(mass=1388, wheelbase=2.68),
+    chassis_codes={"NE"},
   )
-  SKODA_SUPERB_MK3 = VolkswagenMQBPlatformConfig(  # Chassis 3V/NP
+  SKODA_SUPERB_MK3 = VolkswagenMQBPlatformConfig(
     [VWCarDocs("Škoda Superb 2015-22")],
     VolkswagenCarSpecs(mass=1505, wheelbase=2.84),
+    chassis_codes={"3V", "NP"},
   )
 
 
+def match_fw_to_car_custom(live_fw_versions, vin, offline_fw_versions) -> set[str]:
+  candidates = set()
+
+  # Sanity check that the radar FW is likely Volkswagen
+  for ecu, gateway_types in GATEWAY_TYPES.items():
+    radar_fw = live_fw_versions.get(ecu[1:], [])
+    if not len(radar_fw):
+      return set()
+
+    for fw in radar_fw:
+      match = FW_PATTERN.match(fw)
+      if match is None or match.group('gateway') not in gateway_types:
+        return set()
+
+  # Check the WMI and chassis code to determine the platform
+  wmi = vin[:3]
+  if wmi not in WMI:
+    return set()
+
+  chassis_code = vin[6:8]
+  if CHASSIS_CODE_PATTERN.match(chassis_code) is None:
+    return set()
+
+  for platform in CAR:
+    if chassis_code in platform.config.chassis_codes:
+      candidates.add(platform)
+
+  return {str(c) for c in candidates}
+
+
+# Volkswagen uses the VIN WMI and chassis code to match in the absence of the comma power,
+# as we lose too many ECUs to reliably identify the vehicle
+CHASSIS_CODE_PATTERN = re.compile('[A-Z0-9]{2}')
+# TODO: determine the unknown groups
+FW_PATTERN = re.compile(b'\xf1\x87(?P<gateway>[0-9][0-9A-Z]{2})(?P<unknown>[0-9][0-9A-Z][0-9])(?P<unknown2>[0-9A-Z]{2}[0-9])([A-Z0-9]| )')
```

Comment:
@jyoung8607 not a blocker, but would be good to add if you know what the unknown part/parts are